Multisig/UI/Settings/OwnerKeyManagement/AddKeyAsNewOwner/AddOwnerFlowFromSettings.swift

```diff
@@ -64,6 +64,7 @@ class AddOwnerFlowFromSettings: AddOwnerFlow {
             trackingEvent: .addOwnerSpecifyName
         ) { [unowned self] name in
             newAddressName = name
+            AddressBookEntry.addOrUpdate(newOwner!.checksummed, chain: safe.chain!, name: name)
```

Comment:
I'm a bit unsure if this is the right fix for the issue. Can we pass on the name for that address to further screens to display it in Review or if it's added here, then remove it when the flow is cancelled and the name was added?

This is my view of this change:

pros:
 - if the flow is unfinished (cancelled) and then repeated, the name will be shown in the UI already
 - the name is shown on review screen automatically

cons:
  - if the flow is cancelled, then there's new entry in address book which was not used, maybe it would be a mistake to have it then?

Reply:
The latest implementation saves the address book entry in the success screen.